src/lib.rs

```diff
@@ -479,6 +507,110 @@ impl DoubleArrayAhoCorasick {
         }
     }
 
+    /// Serializes the automaton into the output stream.
```

Comment:
This function generates a huge amount of data, so both serialize() and deserialize() should be hidden from users.
Please add `#[doc(hidden)]`